src/states/node.rs

```diff
@@ -1247,51 +1285,30 @@ impl Node {
         self.send_routing_message(response_msg)
     }
 
-    // Received by Y; From A -> Y, or from any node to one of its bucket addresses.
-    fn handle_get_close_group_request(&mut self,
-                                      src: Authority<XorName>,
-                                      dst_name: XorName,
-                                      message_id: MessageId)
-                                      -> Result<(), RoutingError> {
-        // If msg is from ourselves, ignore it.
-        if src.name() == self.name() {
+    fn handle_section_update(&mut self,
+                             prefix: Prefix<XorName>,
+                             members: Vec<PublicId>)
+                             -> Result<(), RoutingError> {
+        trace!("{:?} Got section update for {:?}", self, prefix);
+        // Filter list of members to just those we don't know about:
+        let members = if let Some(section) = self.peer_mgr.routing_table().section_ref(&prefix) {
+            let f = |id: &PublicId| !(section.is_member(id.name()) || section.is_needed(id.name()));
+            members.into_iter().filter(f).collect_vec()
```

Comment:
Wouldn't it be easier to do all this filtering inside `mark_needed` itself (and return `true` there if the entry was in fact new)?
Also, we don't really need to get the section, we could just use `RoutingTable::has` for this, I thin.